npm_and_yarn/lib/dependabot/npm_and_yarn/version_selector.rb

```diff
@@ -0,0 +1,40 @@
+# typed: true
+# frozen_string_literal: true
+
+require "dependabot/shared_helpers"
+
+module Dependabot
+  module NpmAndYarn
+    class VersionSelector
+      # For limited testing, allowing only specific versions defined in engines in package.json
+      # such as "v20.8.7", "20.1.2", "V20.21.2",
+      NODE_ENGINE_SUPPORTED_REGEX = /^(v|V?)\d*(?:\.\d*\.\d*)?$/
```

Comment:
This doesn't quite match your comment above. It will match `20` and `20.0.0`, but not `20.0`.

Reply:
@JamieMagee , thanks for noticing , fixed.